### ISSUE TITLE
Dump: Block styles, folding, etc for multi-line strings

### ISSUE THREAD

**Issue by eforbus** [perlpunk/YAML-PP-p5#3]
I'd love to see support for block notations, literal style, folded sytle, etc  added.. Or maybe it is and I just can't figure out how to make it work (especially since it seems like Emitter.pm  has some handling for this starting at around line 356).

```yaml
# block notation (newlines become spaces)
content:
  Arbitrary free text
  over multiple lines stopping
  after indentation changes...

# literal style (newlines are preserved)
content: |
  Arbitrary free text
  over "multiple lines" stopping
  after indentation changes...

# + indicator (keep extra newlines after block)
content: |+
  Arbitrary free text with two newlines after


# - indicator (remove extra newlines after block)
content: |-
  Arbitrary free text without newlines after it


# folded style (folded newlines are preserved)
content: >
  Arbitrary free text
  over "multiple lines" stopping
  after indentation changes...
```

I would have hoped it would just work for multi-line strings either starting or ending with (or maybe even containing) one or more newlines...

```perl
#!/usr/bin/env perl
use JSON::XS;
use YAML::PP;
use Data::Printer;

my $ypp = YAML::PP->new(boolean => 'JSON::PP', schema => ['JSON']);

my $string = <<END;
This is a multiline string

Testing heredoc and folding in YAML dumping

It doesn't seem to work

END

my $data = {
    properties => {
        foo => {
            in => 'query',
            required => JSON::XS->true,
            description => 'a test',
        },
        bar => {
            in => 'query',
            required => JSON::XS->false
        },
        string => $string
    }
};

print $ypp->dump_string($data);
```

- Current output:

```bash
❯ perl yaml2.pl
---
properties:
  bar:
    in: query
    required: false
  foo:
    description: a test
    in: query
    required: true
  string: "This is a multiline string\n\nTesting heredoc and folding in YAML dumping\n\nIt doesn't seem to work\n\n"

```

- Would love to see

```perl
❯ perl yaml2.pl
---
properties:
  bar:
    in: query
    required: false
  foo:
    description: a test
    in: query
    required: true
  string: |+
    This is a multiline string

    Testing herdoc and folding in YAML dumping

    It doesn't seem to work

```

**Comment by perlpunk** [comment]
Thanks, yeah, this is on my TODO list.
Emitter.pm can output block scalars, but it's not working correctly for many cases yet, so that's why the Dumper just uses double quotes, single quotes or plain currently.

**Comment by perlpunk** [comment]
I uploaded YAML::PP 0.006_002, multiline strings are output as literal block scalars now.
Maybe you can try it out.

**Comment by perlpunk** [comment]
I just released v0.007.
The logic on which style to use when dumping can be tricky. For now I just kept it simple and left out folded style completely.

**Comment by perlpunk** [comment]
No feedback, closing
